lib/Jsrt/JsrtDiag.cpp

```diff
@@ -671,4 +672,19 @@ CHAKRA_API JsDiagEvaluate(
 
     }, false /*allowInObjectBeforeCollectCallback*/, true /*scriptExceptionAllowed*/);
```

Comment:
You did most of the work to implement JsDiagEvaluateUtf8 only work left is to move this to a function taking length as well and call that with wcslen in JsDiagEvaluate and strlen in JsDiagEvaluateUtf8, similar to RunScriptCore?